ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -205,16 +247,19 @@ public void generateFreeTypeElement(final Ds3Type typeEntry) throws IOException
         }
     }
 
-    public Path getRequestOutputPath(final Request request) {
-        return Paths.get(outputDirectory + "/ds3_c_sdk/src/requests/" + request.getNameRootUnderscores() + ".c");
-    }
+    public void generateElementResponseParser(final Type typeEntry) throws IOException {
```

Comment:
This function looks almost identical to generateFreeTypeElement and generateFreeTypeElementPrototype, with the only difference being the name of the template. If this is the case, then maybe these should be refactored into one function that takes a template name as a parameter.